pkg/atlas/meta.go

```diff
@@ -1,10 +1,20 @@
 package atlas
 
 import (
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 )
 
+// PathToProviderJSON is the path to the json file
+var PathToProviderJSON = "providers.json"
+
+// ProvidersFile represents all of the providers that have been set in the provider.json file
+type ProvidersFile struct {
+	Providers []map[string]Provider `json:"providers,omitempty"`
```

Comment:
I would argue that we shouldn't make the user conform to the structure of the undocumented APIs; the structure is non-obvious, unfamiliar and unnecessarily verbose. We also shouldn't override the provider object as received from the API—whilst at the moment we only use the instance sizes, there is a bunch of other information that we will hopefully use in the near future, such as costs and other plan metadata.